### DISCUSSION TITLE
Hide/Group jobs in the status checks list

### DISCUSSION THREAD

**Discussion by donnacello** [community/community#26246] [Actions]
<p>Hi, we have cypress tests running end-to-end tests in parallel and it can be quite a lengthy list. Is there a way to hide these jobs or group them so the PR status check list doesn’t get too populated?<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/8/18b23358ffa1c9d3ff3731292a3d50446bbeb73a.png" data-download-href="/uploads/short-url/3wtfDHYo4ft4Ldwndeypd70P1Ro.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/20868683/181096417-1f4c268c-c131-4462-8298-3a00dfaa1e45.png" alt="image" data-base62-sha1="3wtfDHYo4ft4Ldwndeypd70P1Ro" width="690" height="258" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/8/18b23358ffa1c9d3ff3731292a3d50446bbeb73a_2_690x258.png, https://aws1.discourse-cdn.com/github/original/3X/1/8/18b23358ffa1c9d3ff3731292a3d50446bbeb73a.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/8/18b23358ffa1c9d3ff3731292a3d50446bbeb73a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/8/18b23358ffa1c9d3ff3731292a3d50446bbeb73a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">860×322 42.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Below is a snippet of the workflow from <a href="https://github.com/cypress-io/github-action#parallel" rel="noopener nofollow ugc">cypress github action</a></p>
<pre><code class="lang-auto">name: Parallel Cypress Tests
on: [push]
jobs:
  test:
    name: Cypress run
    runs-on: ubuntu-16.04
    strategy:
      fail-fast: false
      matrix:
        # run 3 copies of the current job in parallel
        containers: [1, 2, 3]
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          record: true
          parallel: true
          group: 'Actions example'
</code></pre>

**Comment by mickeygousset** [comment, marked as answer]
<p>Not that I can think of. Each job is going to show up as a check on the PR. You can click the “Hide All Checks” link to collapse it down, but that is about the best I can think of.</p>

**Reply by leonard-henriquez** [reply]
Do you have any update on this topic? 🙂

**Reply by mickeygousset** [reply]
At this point I still don't see anything on the public roadmap for it.

**Comment by donnacello** [comment]
<p>Thanks so much for your quick response!<br>
Just curious: would this be something that could be easily supported? I imagine a lot of projects may have several jobs triggered per PR.</p>

**Comment by mickeygousset** [comment]
<p>I’m not sure how easily supported it would be, but my bigger question is why would you not want to see all the checks that apply to your PR? Are you asking to just have that area collapsed by default?  Or for something else?</p>
<p>If I said “how would you like to see this feature implemented”, what would you tell me?</p>

**Reply by basickarl** [reply]
An example I have is that I'm using dorny/paths-filter@v2, this just checks to see if files in specific directories have been altered, it's more of a utility job for the actual test jobs. This specific utility job is not prone to fail, ever, so I would also like to hide it or at least have it somewhere more out of view.

Example:
```
jobs:
  changes:
    name: Detect file changes
    runs-on: ubuntu-latest
    outputs:
      packages: ${{ steps.filter.outputs.packages }}
    steps:
    - uses: dorny/paths-filter@v2
      id: filter
      with:
        filters: |
          packages:
            - 'packages/**'
```

**Reply by jamesdh** [reply]
Same exact use-case here. This is a job whose sole purpose is to determine if some other jobs should be run. It never fails, and just creates noise by including it in the Progress Checks list.

**Reply by bsgrigorov** [reply]
same here

**Comment by donnacello** [comment]
<blockquote>
<p>why would you not want to see all the checks that apply to your PR?</p>
</blockquote>
<p>In my team’s case, we have several parallelised tests that are all in the same group of tests (end-to-end tests). In that same group of tests, there is a separate job that runs before all the tests and it verifies if a review environment (created on each pr) is successfully deployed —otherwise the tests will not run. Not all of these necessarily need to show as they are all part of the end-to-end tests.</p>
<blockquote>
<p>Are you asking to just have that area collapsed by default? Or for something else?</p>
</blockquote>
<p>Collapsed by default actually sounds like a good idea!<br>
If you look at what we see here (screenshot below), there are only 5 main “checks” we want to keep track of: unit tests, security tests, end-to-end tests, codeclimate, and review environment creation.  However, because of the parallel tests you would not be able to see everything at a quick glance.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/9/990910c14187ac4c2b782680b12abf678f85e7f2.png" data-download-href="/uploads/short-url/lPOtufXQzcdjsoRmwLxbaQRPGkW.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/20868683/181096420-59fb2cfb-9bf0-42d2-9fca-dedc51ba1552.png" alt="image" data-base62-sha1="lPOtufXQzcdjsoRmwLxbaQRPGkW" width="517" height="190" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/9/990910c14187ac4c2b782680b12abf678f85e7f2_2_517x190.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/9/990910c14187ac4c2b782680b12abf678f85e7f2_2_775x285.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/9/990910c14187ac4c2b782680b12abf678f85e7f2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/9/990910c14187ac4c2b782680b12abf678f85e7f2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">861×317 40.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<blockquote>
<p>If I said “how would you like to see this feature implemented”, what would you tell me?</p>
</blockquote>
<p>On top of your idea of some collapsible element, grouping/collapsing all related jobs into its workflow would be awesome. Maybe it’s something we can specify in the workflow file? (probably not a good solution but just want to throw some ideas in :grin: )</p>
<pre><code class="lang-auto">name: E2E Tests
on: [pull_request]
pr_check_view: collapsed  &lt;--- [collapsed, hidden, display_all_jobs]
jobs: [...]
</code></pre>

**Comment by mickeygousset** [comment]
<p>This is good feedback.  I’ll post it internally and maybe it will bubble up to the backlog.</p>

**Reply by ATOM594** [reply]
Hi @mickeygousset 👋 

How did it go?

**Comment by donnacello** [comment]
<p>hi <a class="mention" href="/mickeygousset">@mickeygousset</a> , are you able to post any update here please? thank you!</p>

**Comment by michaelhays** [comment]
<p>Adding on that an option to group or even hide status checks for a given job would be extremely useful. We also run our Cypress end-to-end tests in parallel, so that they’re split across 8 machines:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/d/3df9e7892a27d10a83b2e851b16d9ddae42642b1.png" data-download-href="/uploads/short-url/8QgxH9ojDUOQLsecWRrV8DKGgSJ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/6445661/181096423-1ff13f58-eb9a-4790-8700-1c3d5c5a4d8f.png" alt="image" data-base62-sha1="8QgxH9ojDUOQLsecWRrV8DKGgSJ" width="690" height="343" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3df9e7892a27d10a83b2e851b16d9ddae42642b1_2_690x343.png, https://aws1.discourse-cdn.com/github/original/3X/3/d/3df9e7892a27d10a83b2e851b16d9ddae42642b1.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/d/3df9e7892a27d10a83b2e851b16d9ddae42642b1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3df9e7892a27d10a83b2e851b16d9ddae42642b1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">922×459 49.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>GitHub automatically creates a new status check for each machine, but we’re really only interested in seeing whether all of the tests passed or failed. In our case, we have a status check set up directly with Cypress that links to the Cypress dashboard (that’s the “cypress: default-group — 2 tests failed” in the image), so the 8 checks created by that job aren’t useful to us.</p>
<p>I’d love to be able to hide (or group) these, e.g.:</p>
<pre><code class="lang-nohighlight">jobs:
  e2e:
    name: Cypress end-to-end tests
    runs-on: ubuntu-latest
    checks: none  # or group, or all (default)
    ...
</code></pre>

**Comment by ggbecker** [comment]
<p>This is definitely something Github should look into it.</p>

**Comment by chasemgray** [comment]
We have 80 parallel machines running, so showing each one in the github checks makes it very hard to use.

**Comment by tinu-doit-intl** [comment]
@mickeygousset any plans to add this feature? We are also facing the same issue when we run the e2e cypress tests in parallel.

**Comment by LeviPesin** [comment]
Would be great to see this implemented.

**Comment by rene-leanix** [comment]
Here's a quick & dirty bookmarklet to toggle successful or skipped checks in the list:

```js
javascript:[...document.querySelectorAll('.merge-status-list .merge-status-item')].filter(el => el.querySelector('.octicon-check, .octicon-skip')).forEach(el => el.hidden = !el.hidden)
```

This may stop working anytime GitHub changes the layout or the CSS classes of that part of their app.

We're working in a mono-repo with 110 checks, so it's extremely frustrating having to scroll and look for the one failed check all the time. Time for this feature to get implemented, @github-staff!

**Reply by ui-kane-lin** [reply]
hi, rene. How to use this script?

**Comment by jamesdh** [comment]
We really need this. Any developer/team that readily takes advantage of GitHub Actions will encounter this problem eventually. The need for it is only going to grow.

**Comment by laeubi** [comment]
https://github.com/EnricoMi/publish-unit-test-result-action#support-fork-repositories-and-dependabot-branches requires to upload the event file, of course this is nothing we want/need to show to the user and is just confusing. So a simple flag on the job 
`show_status: true/false` would be fine.

For Matrix jobs in general, it would be great if only one item is shown for all matrix elements and the can be expanded if needed e.g. lets say I have a Workflow name "Run Buld" with a matrix of jobs "java 8", "java 11", "java 17"

The Github could show one check ("Run Buld") that simply aggregates the status of all childs ("java 8", "java 11", "java 17"), and there is an icon to expand the childs (or one actually can just jump to the job itself)

**Comment by s0undt3ch** [comment]
We're also interested in something, preferably, grouping...

![image](https://user-images.githubusercontent.com/300048/216963988-7ba66da8-0732-4253-a69e-a191320a8dff.png)

**Comment by blasco** [comment]
Same situation here! Many workflows trigger a couple of jobs and currently it is a mess to look at them at the status check report

**Comment by albertodebortoli** [comment]
Same here: we have a job generating the matrix for the subsequent jobs.
Nothing tragic, but surely hiding the first job would be nice.

<img width="614" alt="Screenshot 2023-06-19 at 19 11 31" src="https://github.com/community/community/assets/653112/11b3869b-57e8-4354-9c9f-5db11be51012">

**Comment by datYori** [comment]
same here, when using things like https://github.com/marketplace/actions/changed-files you don't really care about showing the results of this job in your Checks

**Comment by mihaigalos** [comment]


**Reply by magus424** [reply]
This appears to be asking for a way to hide jobs that ran, not skipped jobs. Not quite a dupe. (this one's also older?)

**Comment by magus424** [comment]
I'd also like a way to do this. I have a shared step in some workflows that just builds some values for use in later steps, so it isn't really a "check" on the PR. If one could set something like `is_check: false` or similar it would be handy.

**Comment by a8trejo** [comment]
At the very least for [matrix strategies](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy) this should be an option, since they are the **same** job, 2024 and still waiting for this 🥲.

**Comment by scottwithbanner** [comment]
Same problem for my team — cypress run across a matrix job leaves us with 200+ checks in the PR. Would love to have an option to consolidate all the jobs from a workflow into a single check on the PR

**Comment by simplypixi** [comment]
Same issue here. Some jobs can be just utils jobs, so showing their result as an extra check is redundant.

**Comment by KevinMind** [comment]
Same issue. I agree that the basic use case is to ensure that jobs which run a matrix strategy should only display a single check in the checklist UI. You already have a component for it in showing the progress of a task list for an issue. This circle progress indicator would be perfect.

Next level UX would be to be able to expand the list if you wanted to see individual child jobs.. definitely not a need to have.

The details link can just link to the first job as you end up seeing all the jobs in the side bar anyway.

Who at github wants to give me a codespace to github's webapp and I can ship this in a week?

<img width="120" alt="image" src="https://github.com/community/community/assets/19595165/826df734-41b4-49c6-aada-1d2fa9ad0faf">

**Comment by BevanR** [comment]
This is probably our biggest pain-point with GHA. On one project we actually use a `workflow_run` trigger instead of a `push` trigger _just_ to avoid the commit statuses of our Cypress parallel test workers showing up. The workflow has to jump through a lot of hoops to make that work. Grouping or omitting them would be so much easier.

1. They all have the same name format; "Cypress worker (1)" etc, so it wouldn't even need any config in GH; Just a check for jobs whose name starts with the same first ~10 chars would do.
1. Grouping matrix workers would also be fine; better perhaps.
2. Another simple solution would be a flag on the job in the workflow to indicate that the job status does _not_ affect the commit status, and therefore omit it from the default view and commit status.

🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏 🙏

**Comment by bxt** [comment]
An alternative would be to show the failing matrix jobs on top, like it is done with normal status checks. I would love to see something implemented, as this means a lot of manual workflow calling instead of relying on the build-in functionality.

**Comment by ryan-mcginty-alation** [comment]
Would really love to see this feature as well

**Comment by rrlamichhane** [comment]
let's make this happen please

**Comment by solvingj** [comment]
@mickeygousset and/or @donnacello would you please consider marking this question as un-answered so that it is more easily found, and can be treated as an active request for a major new feature.  All the context is here, evidence of people with the same problem needing the same features.

**Comment by lkfortuna** [comment]
Hi there from Actions team-  there is a ship coming near the end of september from the Pull requests team for a new merge box that groups jobs together by status. Please stay tuned!

**Reply by solvingj** [reply]
That sounds like it would improve the situation a bit for some teams, and I'm glad to hear development is actually occurring. 

However, this won't help unblock my team at all, and I suspect that's true for many others. 

Can you try to suggest to that team that they read feedback left here in this thread, so the feature might be extended/improved further after this upcoming release?  And maybe let us know if they say they will read posts here?

Specific request from our team:
1.  For matrix builds and github checks
- make "group-github-checks-by" a configurable field
- the "group by status" sounds sensible for some teams
- we want to group our matrix builds by "job name", which also seems sensible
- i can imagine other ways people would want to group checks

2.  Make Hide skipped checks a dedicated option
- Regardless of anything else, users will continue to request and ask for hiding skipped checks
- This applies to both checks skipped with "if" and with "on.push.paths"
- Whatever it takes, make it a workflow or job level checkbox so that these can be hidden from github checks

3. Other closely-related Github discussion
- This one is for sure similar https://github.com/orgs/community/discussions/4324
- Any development around Matrix builds and github checks should really be considering all related requests together
- https://github.com/orgs/community/discussions?discussions_q=is%3Aopen+matrix

**Reply by essen** [reply]
Looking forward to this update. We have close to 400 jobs, mostly due to heavy parallelization of tests. In general we don't care about successful jobs, we only care about the failed ones. But having the list of failed tests be grouped under their top-level job would be a plus for clarity.

A convenient link to the workflow(s) summary page would be enough to get more details if we need to check successful jobs as well.

**Reply by michaelhays** [reply]
Any update on that ship @lkfortuna?

**Reply by anikettt2499** [reply]
It would be a nice addition. Any updates?

**Reply by jandrade** [reply]
Do you have any updates on this release? this would be really helpful for similar reasons like the Cypress case described in this discussion.

**Comment by pgodinhofortis** [comment]
You can implement this in different ways:
- A call to GitHub API to delete the check
- A setting on the job for it not to be added as a check

I wish this was looked into with more care, it really bothers us when there are many checks.

**Reply by BevanR** [reply]
Which is the setting on the job that prevents it being added as a check?

**Reply by pgodinhofortis** [reply]
It doesn't exist 😢 I am suggesting simple ways this could be rolled out. 

It is probably a week worth of work but will take years for the actions team to do it.

**Reply by BevanR** [reply]
Ah! I thought you were suggesting workarounds

**Comment by nickgovea** [comment]
Adding to this thread to keep it alive... Would love to see a feature for this implemented. Like some folks, we have Cypress tests that run across multiple machines, it would make more sense to show one parent job for matrix jobs in the Checks section rather than bloating it up with all the parallel jobs as there can be many.

**Comment by liambutler** [comment]
Doesn't just affect Cypress tests; I'm also seeing this repetitive job output with sharded Jest and Playwright tests. Commenting in the hope that more folks find this thread.

**Comment by ui-kane-lin** [comment]
We need this!

**Comment by bernhardkaindl** [comment]
@lkfortuna: Here too!

**Comment by srittau** [comment]
Grouping would also help to distinguish between the matrix parameters in the sidebar of the "Checks" tab. For example, in typeshed after shortening the job names, we can distinguish between the various OSs, but the Python version still remains a mystery:

![image](https://github.com/user-attachments/assets/dc21ae36-38b9-47b6-890b-30e52ea4b748)

Having a group "pyright: Run test cases" with individual entries that just list the matrix parameters (OS, Python version in this case) would be a tremendous help.

**Comment by raffaelsouza-hotmart** [comment]
Any news about this feature?

**Comment by rairulyle** [comment]
Any update on this? This should not be marked as completed.

**Comment by sovdeeth** [comment]
I'd also like to see a feature like this. I'm trying to parallelize tests since sequential runtimes were hitting 10 minutes, but the number of various actions in the check window makes it borderline unusable apart from "everything ok" and "something failed". Grouping would be a great feature.

**Comment by dwjohnston** [comment]
I want to propose an interim solution that would improve the experience a lot. 

Add a direct link to the 'summary' of the workflow execution from the status checks panel. 
<img width="1560" height="672" alt="image" src="https://github.com/user-attachments/assets/11205c69-88aa-4499-9a6b-b931ad57ebad" />


If we can get there in a single click, then the flowchart diagram  is pretty handy in showing what has happened with the run: 

<img width="1304" height="530" alt="image" src="https://github.com/user-attachments/assets/811be36f-41f9-45c2-9a9d-72d406659f12" />

**Reply by dwjohnston** [reply]
TamperMonkey script for the above: 

```
// ==UserScript==
// @name         GitHub Actions Run Link Inserter (general)
// @version      0.2
// @description  Add a link to the summary section of a workflow execution.
// @author       dwjohnston
// @match        https://github.com/*
// @grant        none
// @run-at       document-idle
// ==/UserScript==

(function () {
    'use strict';

    console.log("Tampermonkey script to add summary link to the github status check");

    function isPrPage(url){
        return !!(url.pathname.match(/\/pull\/\d+$/));
    }


    function runScriptIfOnPullRequestPage(url){


        if(!isPrPage(url)){
            return;
        }

        console.log("Pull request page detected, attemping insert....");

        let interval;
        const MAX_ATTEMPTS = 10;
        let attemptNumber =0;

        function attemptInsert() {

            console.log(`Attempting insert #${attemptNumber}`);

            const target = document.querySelector('section[aria-label="Checks"]>div:first-child>div:first-child');
            if(!target){
                return false;
            }

            // I'm assuming that the first link to a `/...` is good enough.
            const firstRelativeLink = document.querySelector(`section[aria-label="Checks"] a[href^="/"`);
            if(!firstRelativeLink) {
                return false;
            }

            const url = firstRelativeLink.href;
            const newUrl = url.replace(/\/job\/\d+/, 'summary')


            const link = document.createElement("a");
            link.setAttribute("href", newUrl);
            link.innerHTML = "Summary"
            link.setAttribute("style", "z-index:1000; display: block; padding: 1em;");

            target.prepend(link);
            return true;

        }

        const navigateAway = (event) => {

            clearInterval(interval);
            navigation.removeEventListener("navigate",navigateAway);
        }

        navigation.addEventListener("navigate", navigateAway);



        interval = setInterval(() => {
            if(attemptNumber >= MAX_ATTEMPTS) {
                clearInterval(interval);
            }

            const result = attemptInsert();
            if(result){
                clearInterval(interval);
            }

            attemptNumber++;

        }, 1000);
    }



    navigation.addEventListener("navigate", (e) => {
        const url = new URL(e.destination.url);
        runScriptIfOnPullRequestPage(url);
    });

})();
```

<img width="1048" height="314" alt="image" src="https://github.com/user-attachments/assets/df2da0d4-32d4-4563-b36d-acbc686acbf1" />
